### ISSUE TITLE
Monochromatic light behaves as a switch instead of a 'fan-like' component over the web view

### ISSUE THREAD

**Issue by cociweb** [esphome/esphome-webserver#15]
Hello,

A monochromatic (dimmable) light device has a weired behaviour over the inbuilt esphome webserver. The dimmable device visibilie as a pure ON/OFF switch. On the web interface It should be visible (and controllable) as a fan-like integration: on/off state + dimmed value in percentage.

There is no issue over the useblility over the HA integration/api. The monochromatic component itself (except the web-ui) works like a charm.

### Example YAML snippet

```yaml
light:
  - platform: monochromatic
    name: "Világítás"
    id: light
    output: light_out
    default_transition_length: 1.0s
    
fan:
  - platform: speed
    name: "Világítás-fan"
    id: fan
    output: light_out

output:
  - platform: esp8266_pwm
    id: light_out
    pin: GPIO14
    frequency: 200 Hz
    zero_means_zero: true
    min_power: 0.00
    max_power: 1.00
```

### Log

```txt
02:06:50	[D]	[light:035]	'Világítás' Setting:
02:06:50	[D]	[light:046]	  State: ON
02:06:50	[D]	[light:084]	  Transition length: 1.0s
02:06:50	[D]	[light:035]	'Világítás' Setting:
02:06:50	[D]	[light:046]	  State: OFF
02:06:50	[D]	[light:084]	  Transition length: 1.0s

....
02:20:35	[D]	[fan:021]	'Világítás-fan' - Setting:
02:20:35	[D]	[fan:024]	  State: OFF
02:20:35	[D]	[fan:091]	'Világítás-fan' - Sending state:
02:20:35	[D]	[fan:092]	  State: OFF
02:20:35	[D]	[fan:094]	  Speed: 100
02:20:43	[D]	[fan:021]	'Világítás-fan' - Setting:
02:20:43	[D]	[fan:024]	  State: ON
02:20:43	[D]	[fan:091]	'Világítás-fan' - Sending state:
02:20:43	[D]	[fan:092]	  State: ON
02:20:43	[D]	[fan:094]	  Speed: 46
```

### Screenshot

![image](https://user-images.githubusercontent.com/1935413/186548671-03a4b3c2-c2c4-4346-bc92-29968adeef8e.png)

**Comment by rob121** [comment]
For anyone who is coming here with this issue - as a work around you can create a number input slider and then on_value update the brightness of monochromatic light

```
number:
  - platform: template
    name: "Brightness"
    id: "brightness_1"
    step: .10
    restore_value: true
    min_value: 0
    max_value: 1
    optimistic: true
    mode: slider
    on_value:
      then:
      - light.turn_on:
          id: light_1
          brightness: !lambda "return x;"
```

**Comment by cociweb** [comment]
Hello @rob121,
Your suggestion works like a charm! The webgui has a slider now (the light entity moved as internal, to hide from the webgui). Unfortunatelly, since the light entity is internal the home assistant not able to control it (-> light automations/blueprints need to be reworked for numbers: never ending story)  Anyway, as you mentioned it is a workaround only, and it works.  Thank's @rob121 once again!

However,  it does not solve the original issue.

**Comment by jesserockz** [comment]
Closed with #19
